### ISSUE TITLE
8B还是18B？

### ISSUE THREAD

**Issue by lucasjinreal** [OpenSenseNova/SenseNova-U1#94]
为什么这么大体积非得说8B？

**Comment by yl-1993** [comment]
@lucasjinreal 感谢您的问题，这个模型由于是Mixture-of-Transformer的结构，取名依循了[BAGEL-7B-MoT](https://huggingface.co/ByteDance-Seed/BAGEL-7B-MoT)所创立的规范。因此实际上，SenseNova-U1-8B-MoT包括8B理解和8B生成的参数。

为了避免命名带来的误解，PR https://github.com/OpenSenseNova/SenseNova-U1/pull/102 提供了一个查看模型参数的脚本，U1-8B-MoT总的参数量是17.552B，差不多是您提到的18B。PR中也在README的Models部分做了说明。

同时，为了更好呈现模型结构，我们也根据模型中参数名称的解析，列出详细的模型参数统计：

```
Model: sensenova/SenseNova-U1-8B-MoT
Load dtype:   bfloat16
Total params: 17.552B
Total memory: 35.105GB (bfloat16)
---------------------------------------------------------------------
group                              params memory (bfloat16)      ratio
---------------------------------------------------------------------
generation_transformer             8.186B         16.373GB     46.64%
understanding_transformer          8.121B         16.243GB     46.27%
shared                             1.245B          2.489GB      7.09%
---------------------------------------------------------------------
Pathway breakdown (shared counted in both):
---------------------------------------------------------------------
pathway                            params memory (bfloat16)      ratio
---------------------------------------------------------------------
understanding pathway              9.366B         18.732GB     53.36%
generation pathway                 9.431B         18.862GB     53.73%

---------------------------------------------------------------------
Members of group 'shared' (2 params, 1.245B total, 2.489GB @ bfloat16)
---------------------------------------------------------------------
param name                                                  numel    dtype
---------------------------------------------------------------------
language_model.model.embed_tokens.weight                 622.330M bfloat16
language_model.lm_head.weight                            622.330M bfloat16
```

第一段表示**模型的参数量** (每个参数物理上只算一次，加起来100%)
- understanding_transformer ≈ 8.12B (46%)
- generation_transformer ≈ 8.19B (47%)
- shared ≈ 1.25B (7%)

第二段表示**模型forward时候经过的参数量** (由于有共同经过的部分，加起来会超过100%）
- 跑理解任务时：图像走 vision_model → token 序列经过 embed_tokens → LLM 走非 _mot_gen"分支 → lm_head 出 logits → 总激活参数 ≈ understanding_transformer + shared ≈ 9.37B (53%)
- 跑图生任务时(以比较复杂的单轮thinking interleave来考虑)：图像走 fm_modules.vision_model_mot_gen + text prompt 经过 embed_tokens → LLM 走 _mot_gen 分支 →  lm_head + fm_modules.fm_head → 总激活参数 ≈ generation_transformer + shared ≈ 9.43B (54%)

**Comment by lucasjinreal** [comment]
有没有计划缩小一点，小一点的显卡也能装得下

**Comment by yl-1993** [comment]
@lucasjinreal 嗯嗯，小模型我们是认同的，也很感谢这个建议。

我们前期其实做过一些小模型的尝试。在 [NEO-Unify](https://huggingface.co/blog/sensenova/neo-unify) 里提到过 2B-MoT 的早期实验。受限于总体的训练资源和对理解生成效果的考量，没有像当前 8B-MoT 版本一样进行充分训练，所以暂时没有作为正式模型发布。

如果社区对小显卡友好的版本呼声比较高，我们会重新评估更小规格（例如 2B 级别）模型的训练和发布计划。

**Comment by shaohk** [comment]
@yl-1993  可以支持Mac吗？
